File: yyxarch/src/main/java/com/yyxnb/yyxarch/http/client/HttpClient.java
package com.yyxnb.yyxarch.http.client;

import okhttp3.OkHttpClient;

/**
 * okHttp client
 */

public class HttpClient {

    private static HttpClient instance;

    private OkHttpClient.Builder builder;

    public HttpClient() {
        builder = new OkHttpClient.Builder();
    }

    public static HttpClient getInstance() {

        if (instance == null) {
            synchronized (HttpClient.class) {
                if (instance == null) {
                    instance = new HttpClient();
                }
            }
        }
        return instance;
    }


    public OkHttpClient.Builder getBuilder() {
        return builder;
    }

}

File: yyxarch/src/main/java/com/yyxnb/yyxarch/http/client/HttpClient.kt
package com.yyxnb.yyxarch.http.client

import okhttp3.OkHttpClient

/**
 * okHttp client
 */

class HttpClient {

    val builder: OkHttpClient.Builder

    init {
        builder = OkHttpClient.Builder()
    }

    companion object {

        private var instance: HttpClient? = null

        fun getInstance(): HttpClient {

            if (instance == null) {
                synchronized(HttpClient::class.java) {
                    if (instance == null) {
                        instance = HttpClient()
                    }
                }
            }
            return instance!!
        }
    }

}
